Program panics when using multiple references in compound document

I have written a test project to use.
Please find the code at https://github.com/tanmayambre/testapi2go.git

As an example document author has two refernced documents. "blog" and "book" 
I implmented functions for jsonapi.EditToManyRelations interface 
If I implement DeleteToManyIDs(name string, IDs []string) error the program compiles but running it causes it to panic. 
The stacktrace is at the end of the description  and also in the test project.

If I use just single document as a reference it works completely fine. 
If I remove DeleteToManyIDs function keeping the tworefernces in document it doesn't throw a panic. 

Please help I cant find if there is an obvious mistake if there is any.

```
Stacktrace.

panic: a handle is already registered for path ''/v1/authors/:id/links/:name'

goroutine 1 [running]:


github.com/julienschmidt/httprouter.(*node).addRoute(0xc208046960, 0xc2080a3756, 0x5, 0xc2080a3720)
	/home/bellrock1/gocode/src/github.com/julienschmidt/httprouter/tree.go:190 +0xc7d
github.com/julienschmidt/httprouter.(*Router).Handle(0xc2080cc150, 0x73b0f0, 0x4, 0xc2080a3740, 0x1b, 0xc2080a3720)
	/home/bellrock1/gocode/src/github.com/julienschmidt/httprouter/router.go:229 +0x1fd
github.com/julienschmidt/httprouter.(*Router).POST(0xc2080cc150, 0xc2080a3740, 0x1b, 0xc2080a3720)
	/home/bellrock1/gocode/src/github.com/julienschmidt/httprouter/router.go:188 +0x59
github.com/manyminds/api2go.(*API).addResource(0xc208046dc0, 0x7fb3672faea8, 0xc208058240, 0x7fb3672fae68, 0xc208038258, 0xc20807de30)
	/home/bellrock1/gocode/src/github.com/manyminds/api2go/api.go:329 +0xe42
github.com/manyminds/api2go.(*API).AddResource(0xc208046dc0, 0x7fb3672faea8, 0xc208058240, 0x7fb3672fae68, 0xc208038258)
	/home/bellrock1/gocode/src/github.com/manyminds/api2go/api.go:373 +0x50
main.main()
	/home/bellrock1/gocode/src/github.com/testapi2go/main.go:15 +0x113
```

--- a/README.md
+++ b/README.md
@@ -202,7 +202,7 @@ will yield
 
 You can also use `jsonapi.MarshalToJSONWithURLs` to automatically generate URLs for the rest endpoints that have a
 version and BaseURL prefix. This will generate the same routes that our API uses. This adds `self` and `related` fields
-for relations inside the `links` object.
+for relations inside the `relationships` object.
 
 Recover the structure from above using
 
@@ -336,12 +336,12 @@ GET     /v1/posts/<id>
 PATCH   /v1/posts/<id>
 DELETE  /v1/posts/<id>
 GET     /v1/posts/<id>/comments            // fetch referenced comments of a post
-GET     /v1/posts/<id>/links/comments      // fetch IDs of the referenced comments only
-PATCH   /v1/posts/<id>/links/comments      // replace all related comments
+GET     /v1/posts/<id>/relationships/comments      // fetch IDs of the referenced comments only
+PATCH   /v1/posts/<id>/relationships/comments      // replace all related comments
 
 // These 2 routes are only created for to-many relations that implement EditToManyRelations interface
-POST    /v1/posts/<id>/links/comments      // Add a new comment reference, only for to-many relations
-DELETE  /v1/posts/<id>/links/comments      // Delete a comment reference, only for to-many relations
+POST    /v1/posts/<id>/relationships/comments      // Add a new comment reference, only for to-many relations
+DELETE  /v1/posts/<id>/relationships/comments      // Delete a comment reference, only for to-many relations
 ```
 
 For the last two generated routes, it is necessary to implement the `jsonapi.EditToManyRelations` interface.
@@ -426,7 +426,7 @@ The IDs of a relationship can be fetched by following the `self` link of a relat
 of a result. For the posts and comments example you could use the following generated URL:
 
 ```
-GET /v1/posts/1/links/comments
+GET /v1/posts/1/relationships/comments
 ```
 
 This would return all comments that are currently referenced by post with ID 1. For example:
@@ -434,7 +434,7 @@ This would return all comments that are currently referenced by post with ID 1. 
 ```json
 {
   "links": {
-    "self": "/v1/posts/1/links/comments",
+    "self": "/v1/posts/1/relationships/comments",
     "related": "/v1/posts/1/comments"
   },
   "data": [
@@ -451,7 +451,7 @@ This would return all comments that are currently referenced by post with ID 1. 
 ```
 
 ### Fetching related resources
-Api2go always creates a `related` field for elements in the `links` object of the result. This is like it's
+Api2go always creates a `related` field for elements in the `relationships` object of the result. This is like it's
 specified on jsonapi.org. Post example:
 
 ```json
@@ -465,7 +465,7 @@ specified on jsonapi.org. Post example:
         "comments": {
           "links": {
             "related": "/v1/posts/1/comments",
-            "self": "/v1/posts/1/links/comments"
+            "self": "/v1/posts/1/relationships/comments"
           },
           "data": [
             {

--- a/api.go
+++ b/api.go
@@ -382,7 +382,7 @@ func (api *API) addResource(prototype jsonapi.MarshalIdentifier, source CRUD, ma
 
 			if _, ok := ptrPrototype.(jsonapi.EditToManyRelations); ok && relation.Name == jsonapi.Pluralize(relation.Name) {
 				// generate additional routes to manipulate to-many relationships
-				api.router.POST(api.prefix+name+"/:id/relationships/:name", func(relation jsonapi.Reference) httprouter.Handle {
+				api.router.POST(api.prefix+name+"/:id/relationships/"+relation.Name, func(relation jsonapi.Reference) httprouter.Handle {
 					return func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
 						err := res.handleAddToManyRelation(w, r, ps, relation)
 						if err != nil {
@@ -391,7 +391,7 @@ func (api *API) addResource(prototype jsonapi.MarshalIdentifier, source CRUD, ma
 					}
 				}(relation))
 
-				api.router.DELETE(api.prefix+name+"/:id/relationships/:name", func(relation jsonapi.Reference) httprouter.Handle {
+				api.router.DELETE(api.prefix+name+"/:id/relationships/"+relation.Name, func(relation jsonapi.Reference) httprouter.Handle {
 					return func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
 						err := res.handleDeleteToManyRelation(w, r, ps, relation)
 						if err != nil {

--- a/api_test.go
+++ b/api_test.go
@@ -22,6 +22,7 @@ type Post struct {
 	Value    null.Float
 	Author   *User     `json:"-"`
 	Comments []Comment `json:"-"`
+	Bananas  []Banana  `json:"-"`
 }
 
 func (p Post) GetID() string {
@@ -43,6 +44,10 @@ func (p Post) GetReferences() []jsonapi.Reference {
 			Name: "comments",
 			Type: "comments",
 		},
+		{
+			Name: "bananas",
+			Type: "bananas",
+		},
 	}
 }
 
@@ -53,6 +58,9 @@ func (p Post) GetReferencedIDs() []jsonapi.ReferenceID {
 	}
 	for _, comment := range p.Comments {
 		result = append(result, jsonapi.ReferenceID{ID: comment.GetID(), Name: "comments", Type: "comments"})
+	}
+	for _, banana := range p.Bananas {
+		result = append(result, jsonapi.ReferenceID{ID: banana.GetID(), Name: "bananas", Type: "bananas"})
 	}
 
 	return result
@@ -81,6 +89,14 @@ func (p *Post) SetToManyReferenceIDs(name string, IDs []string) error {
 		p.Comments = comments
 	}
 
+	if name == "bananas" {
+		bananas := []Banana{}
+		for _, ID := range IDs {
+			bananas = append(bananas, Banana{ID: ID})
+		}
+		p.Bananas = bananas
+	}
+
 	return errors.New("There is no to-many relationship with the name " + name)
 }
 
@@ -88,6 +104,12 @@ func (p *Post) AddToManyIDs(name string, IDs []string) error {
 	if name == "comments" {
 		for _, ID := range IDs {
 			p.Comments = append(p.Comments, Comment{ID: ID})
+		}
+	}
+
+	if name == "bananas" {
+		for _, ID := range IDs {
+			p.Bananas = append(p.Bananas, Banana{ID: ID})
 		}
 	}
 
@@ -106,6 +128,16 @@ func (p *Post) DeleteToManyIDs(name string, IDs []string) error {
 		}
 	}
 
+	if name == "bananas" {
+		for _, ID := range IDs {
+			// find and delete the comment with ID
+			for pos, banana := range p.Bananas {
+				if banana.GetID() == ID {
+					p.Bananas = append(p.Bananas[:pos], p.Bananas[pos+1:]...)
+				}
+			}
+		}
+	}
 	return errors.New("There is no to-manyrelationship with the name " + name)
 }
 
@@ -116,6 +148,9 @@ func (p Post) GetReferencedStructs() []jsonapi.MarshalIdentifier {
 	}
 	for key := range p.Comments {
 		result = append(result, p.Comments[key])
+	}
+	for key := range p.Bananas {
+		result = append(result, p.Bananas[key])
 	}
 
 	return result
@@ -128,6 +163,15 @@ type Comment struct {
 
 func (c Comment) GetID() string {
 	return c.ID
+}
+
+type Banana struct {
+	ID   string `jnson:"-"`
+	Name string
+}
+
+func (b Banana) GetID() string {
+	return b.ID
 }
 
 type User struct {
@@ -437,6 +481,13 @@ var _ = Describe("RestHandler", func() {
 							"related": "/v1/posts/1/comments",
 						},
 					},
+					"bananas": map[string]interface{}{
+						"data": []map[string]interface{}{},
+						"links": map[string]string{
+							"self":    "/v1/posts/1/relationships/bananas",
+							"related": "/v1/posts/1/bananas",
+						},
+					},
 				},
 			}
 
@@ -479,6 +530,13 @@ var _ = Describe("RestHandler", func() {
 							"related": "/v1/posts/2/comments",
 						},
 					},
+					"bananas": map[string]interface{}{
+						"data": []map[string]interface{}{},
+						"links": map[string]string{
+							"self":    "/v1/posts/2/relationships/bananas",
+							"related": "/v1/posts/2/bananas",
+						},
+					},
 				},
 			}
 
@@ -502,6 +560,13 @@ var _ = Describe("RestHandler", func() {
 						"links": map[string]string{
 							"self":    "/v1/posts/3/relationships/comments",
 							"related": "/v1/posts/3/comments",
+						},
+					},
+					"bananas": map[string]interface{}{
+						"data": []map[string]interface{}{},
+						"links": map[string]string{
+							"self":    "/v1/posts/3/relationships/bananas",
+							"related": "/v1/posts/3/bananas",
 						},
 					},
 				},
@@ -641,6 +706,13 @@ var _ = Describe("RestHandler", func() {
 							"links": map[string]interface{}{
 								"self":    "/v1/posts/4/relationships/comments",
 								"related": "/v1/posts/4/comments",
+							},
+						},
+						"bananas": map[string]interface{}{
+							"data": []interface{}{},
+							"links": map[string]interface{}{
+								"self":    "/v1/posts/4/relationships/bananas",
+								"related": "/v1/posts/4/bananas",
 							},
 						},
 					},
@@ -926,7 +998,7 @@ var _ = Describe("RestHandler", func() {
 				"1": &Post{ID: "1", Title: "Hello, World!"},
 			}, false}
 
-			jsonResponse = `{"data":{"attributes":{"title":"Hello, World!","value":null},"id":"1","relationships":{"author":{"data":null,"links":{"related":"/posts/1/author","self":"/posts/1/relationships/author"}},"comments":{"data":[],"links":{"related":"/posts/1/comments","self":"/posts/1/relationships/comments"}}},"type":"posts"}}`
+			jsonResponse = `{"data":{"attributes":{"title":"Hello, World!","value":null},"id":"1","relationships":{"author":{"data":null,"links":{"related":"/posts/1/author","self":"/posts/1/relationships/author"}},"bananas":{"data":[],"links":{"related":"/posts/1/bananas","self":"/posts/1/relationships/bananas"}},"comments":{"data":[],"links":{"related":"/posts/1/comments","self":"/posts/1/relationships/comments"}}},"type":"posts"}}`
 			prettyResponse = `{
     "data": {
         "attributes": {
@@ -940,6 +1012,13 @@ var _ = Describe("RestHandler", func() {
                 "links": {
                     "related": "/posts/1/author",
                     "self": "/posts/1/relationships/author"
+                }
+            },
+            "bananas": {
+                "data": [],
+                "links": {
+                    "related": "/posts/1/bananas",
+                    "self": "/posts/1/relationships/bananas"
                 }
             },
             "comments": {
@@ -1051,6 +1130,13 @@ var _ = Describe("RestHandler", func() {
 							"related": "http://localhost:1337/v0/posts/1/author",
 						},
 					},
+					"bananas": map[string]interface{}{
+						"data": []interface{}{},
+						"links": map[string]interface{}{
+							"self":    "http://localhost:1337/v0/posts/1/relationships/bananas",
+							"related": "http://localhost:1337/v0/posts/1/bananas",
+						},
+					},
 					"comments": map[string]interface{}{
 						"data": []interface{}{},
 						"links": map[string]interface{}{
@@ -1074,6 +1160,13 @@ var _ = Describe("RestHandler", func() {
 						"links": map[string]interface{}{
 							"self":    "http://localhost:1337/v0/posts/2/relationships/author",
 							"related": "http://localhost:1337/v0/posts/2/author",
+						},
+					},
+					"bananas": map[string]interface{}{
+						"data": []interface{}{},
+						"links": map[string]interface{}{
+							"self":    "http://localhost:1337/v0/posts/2/relationships/bananas",
+							"related": "http://localhost:1337/v0/posts/2/bananas",
 						},
 					},
 					"comments": map[string]interface{}{

--- a/examples/crud_example.go
+++ b/examples/crud_example.go
@@ -23,16 +23,16 @@ Create a chocolate with the name sweet
 	curl -X POST http://localhost:31415/v0/chocolates -d '{"data" : [{"type" : "chocolates" , "name" : "Ritter Sport", "taste": "Very Good"}]}'
 
 Create a user with a sweet
-	curl -X POST http://localhost:31415/v0/users -d '{"data" : [{"type" : "users" , "user-name" : "marvin", "links": {"sweets": {"linkage": [{"type": "chocolates", "id": "1"}]}}}]}'
+	curl -X POST http://localhost:31415/v0/users -d '{"data" : [{"type" : "users" , "user-name" : "marvin", "relationships": {"sweets": {"linkage": [{"type": "chocolates", "id": "1"}]}}}]}'
 
 Replace a users sweets
-	curl -X PATCH http://localhost:31415/v0/users/1/links/sweets -d '{"data" : [{"type": "chocolates", "id": "2"}]}'
+	curl -X PATCH http://localhost:31415/v0/users/1/relationships/sweets -d '{"data" : [{"type": "chocolates", "id": "2"}]}'
 
 Add a sweet
-	curl -X POST http://localhost:31415/v0/users/1/links/sweets -d '{"data" : [{"type": "chocolates", "id": "2"}]}'
+	curl -X POST http://localhost:31415/v0/users/1/relationships/sweets -d '{"data" : [{"type": "chocolates", "id": "2"}]}'
 
 Remove a sweet
-	curl -X DELETE http://localhost:31415/v0/users/1/links/sweets -d '{"data" : [{"type": "chocolates", "id": "2"}]}'
+	curl -X DELETE http://localhost:31415/v0/users/1/relationships/sweets -d '{"data" : [{"type": "chocolates", "id": "2"}]}'
 */
 package main
 

